.gitlab-ci.py

```diff
@@ -16,13 +17,18 @@
     python.pytest(),
     python.mypy("gcip"),
     python.bdist_wheel(),
+)
+
+dcc = DockerClientConfig(config_file_path="/kaniko/.docker")
+dcc.add_auth("https://index.docker.io/v1/", username_env_var="DOCKER_USER", password_env_var="DOCKER_LOGIN")
+pipeline.add_children(
     kaniko.execute(
         gitlab_executor_image=kaniko_image,
         image_name="thomass/gcip",
-        enable_push=(PredefinedVariables.CI_COMMIT_TAG or PredefinedVariables.CI_COMMIT_BRANCH == "main"),
-        registry_user_env_var="DOCKER_USER",
-        registry_login_env_var="DOCKER_LOGIN",
+        enable_push=True if PredefinedVariables.CI_COMMIT_TAG or PredefinedVariables.CI_COMMIT_BRANCH == "main" else False,
```

Comment:
Suggestion 1: Why do you want to check if `PredefinedVariables.CI_COMMIT_TAG` is not None? `PredefinedVariables.CI_COMMIT_TAG` is either `None` or a String. No need for `is not None`

Suggestion 2: The expression is correct but, argument `enable_push` takes a `boolean` value. 
The expression won't return a boolean, instead it will return the actual string.

Reply:
As discussed, an thanks than we found a proper solution.
`PredefinedVariables.CI_COMMIT_TAG is not None or PredefinedVariables.CI_COMMIT_BRANCH == "main"`